project-koku/koku
# COST-673, COST-790: GCP markup

GCP markup

**Testing**
1. Create a GCP source
2. Add GCP markup cost model
3. Ingest data and verify cost model is added to the markup field from the COST API

**Test Results**
1.
```
GET /api/cost-management/v1/sources/
HTTP 200 OK
Allow: GET, HEAD, POST
Cache-Control: max-age=0, no-cache, no-store, must-revalidate, private
Content-Type: application/json
Expires: Sun, 10 Jan 2021 05:09:04 GMT
Vary: Accept

{
    "meta": {
        "count": 1
    },
    "links": {
        "first": "/api/cost-management/v1/sources/?limit=10&offset=0",
        "next": null,
        "previous": null,
        "last": "/api/cost-management/v1/sources/?limit=10&offset=0"
    },
    "data": [
        {
            "id": 1,
            "uuid": "06e554bd-d81c-443c-bf9b-a66924c3c3bf",
            "name": "GCP Source",
            "source_type": "GCP",
            "authentication": {
                "credentials": {
                    "project_id": "ascendant-fake-999999"
                }
            },
            "billing_source": {
                "data_source": {
                    "dataset": "ExampleDataset",
                    "table_id": "gcp_billing_export_example_table"
                }
            },
            "provider_linked": true,
            "active": true,
            "current_month_data": true,
            "infrastructure": "Unknown",
            "cost_models": [
                {
                    "name": "GCP Markup",
                    "uuid": "b29d7fb2-e0d7-460f-a7c5-fe467618d18b"
                }
            ]
        }
    ]
}
```
2.
```
GET /api/cost-management/v1/cost-models/
HTTP 200 OK
Allow: GET, POST, HEAD
Cache-Control: max-age=0, no-cache, no-store, must-revalidate, private
Content-Type: application/json
Expires: Sun, 10 Jan 2021 05:08:44 GMT
Vary: Accept

{
    "meta": {
        "count": 1
    },
    "links": {
        "first": "/api/cost-management/v1/cost-models/?limit=10&offset=0",
        "next": null,
        "previous": null,
        "last": "/api/cost-management/v1/cost-models/?limit=10&offset=0"
    },
    "data": [
        {
            "uuid": "b29d7fb2-e0d7-460f-a7c5-fe467618d18b",
            "name": "GCP Markup",
            "description": "test",
            "source_type": "Google Cloud Platform",
            "created_timestamp": "2021-01-10T04:35:59.208446Z",
            "updated_timestamp": "2021-01-10T04:35:59.208482Z",
            "rates": [],
            "markup": {
                "value": "10.0000000000",
                "unit": "percent"
            },
            "source_uuids": [],
            "sources": [
                {
                    "uuid": "06e554bd-d81c-443c-bf9b-a66924c3c3bf",
                    "name": "GCP Source"
                }
            ]
        }
    ]
}
```

3.
```GET /api/cost-management/v1/reports/gcp/costs/
HTTP 200 OK
Allow: GET, HEAD, OPTIONS
Content-Type: application/json
Vary: X_RH_IDENTITY, Accept

{
    "meta": {
        "count": 10,
        "filter": {
            "time_scope_value": "-10",
            "time_scope_units": "day",
            "resolution": "daily"
        },
        "group_by": {},
        "order_by": {},
        "total": {
            "infrastructure": {
                "raw": {
                    "value": 5.081236,
                    "units": "GBP"
                },
                "markup": {
                    "value": 0.5081236,
                    "units": "GBP"
                },
                "usage": {
                    "value": 0.0,
                    "units": "GBP"
                },
                "total": {
                    "value": 5.5893596,
                    "units": "GBP"
                }
            },
            "supplementary": {
                "raw": {
                    "value": 0.0,
                    "units": "GBP"
                },
                "markup": {
                    "value": 0.0,
                    "units": "GBP"
                },
                "usage": {
                    "value": 0.0,
                    "units": "GBP"
                },
                "total": {
                    "value": 0.0,
                    "units": "GBP"
                }
            },
            "cost": {
                "raw": {
                    "value": 5.081236,
                    "units": "GBP"
                },
                "markup": {
                    "value": 0.5081236,
                    "units": "GBP"
                },
                "usage": {
                    "value": 0.0,
                    "units": "GBP"
                },
                "total": {
                    "value": 5.5893596,
                    "units": "GBP"
                }
            }
        }
    },
```
